GCRCatalogs/dc2_object.py

```diff
@@ -366,6 +377,31 @@ def _generate_datasets(self):
 
         return datasets
 
+    @staticmethod
+    def _generate_columns_from_yaml(schema_path):
+        """Return column names based on schema in YAML file
+
+        Args:
+            schema_path (string): <file path> to schema file.
+
+        Returns:
+            The column names defined in the schema.
+
+        Warns:
+            If one or more column names are repeated.
+        """
+
+        with open(schema_path, 'r') as schema_stream:
+            schema = yaml.load(schema_stream)
+
+        if schema is None:
+            warn_msg = 'Schema file "{}" empty.'
+            warnings.warn(warn_msg.format(schema_path))
+            return set()
```

Comment:
I think this should either raise an error, or we modify the logic in Lines 191-197 so that we first run `_generate_columns_from_yaml` and check if the result is empty, and if so, fall back to  `_generate_columns`.